### PR TITLE
Added formula for libspnav

### PR DESCRIPTION
**Pull request by justbuchanan**
Libspnav is a client library for interfacing with 3DConnexion's 3D input devices.

### CODE REVIEW

File: Library/Formula/libspnav.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (DomT4) [inline Library/Formula/libspnav.rb]: Don't need an empty line here.

File: Library/Formula/libspnav.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class Libspnav < Formula
+  desc "A free, compatible alternative for 3Dconnexion's 3D input device drivers and SDK"
```
Comment (DomT4) [inline Library/Formula/libspnav.rb]:
```
* Description is too long. "name: desc" should be less than 80 characters.
Length is calculated as libspnav + desc. (currently 90)
* Please remove the indefinite article "A" from the beginning of the description
```

File: Library/Formula/libspnav.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  def install
+    args = %W[
+      --disable-debug
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --prefix=#{prefix}
+    ]
+
+    if build.without? "x11"
```
Comment (DomT4) [inline Library/Formula/libspnav.rb]: Can dump this on one line:

```ruby
args << "--disable-x11" if build.without? "x11"
```

**Comment by justbuchanan** [comment]
Hi @DomT4, thanks for reviewing.  I implemented your suggestions, so let me know if you see anything else.

File: Library/Formula/libspnav.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Libspnav < Formula
+  desc "Client library for connecting to 3Dconnexion's 3D input devices"
+  homepage "http://spacenav.sourceforge.net/index.html"
+  url "https://downloads.sourceforge.net/project/spacenav/spacenav%20library%20%28SDK%29/libspnav%200.2.3/libspnav-0.2.3.tar.gz"
+  sha256 "7ae4d7bb7f6a5dda28b487891e01accc856311440f582299760dace6ee5f1f93"
+  option "with-x11", "Enable support for sending mouse events through the x11 protocol"
```
Comment (bfontaine) [inline Library/Formula/libspnav.rb]: Please leave an empty line between the `sha256` and the `option`.

Comment (justbuchanan) [inline Library/Formula/libspnav.rb]: Done.

File: Library/Formula/libspnav.rb (hunk 1 above)
Comment (bfontaine) [inline Library/Formula/libspnav.rb]: @DomT4 I think GitHub messed up with the diff and your comment isn’t relevant anymore. Is it?

Comment (DomT4) [inline Library/Formula/libspnav.rb]: No, It's cool now. There was an empty line at the start of the file, and apparently when that was fixed Git decided to grab the next empty line and apply my comment there.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by mikemcquaid** [comment]
For future reference the preferred commit message format for new formulae is `libspnav 0.2.3 (new formula)`.

**Pull request closed without merging**